src/main/java/WayofTime/bloodmagic/compat/waila/provider/DataProviderBloodTank.java

```diff
@@ -0,0 +1,78 @@
+package WayofTime.bloodmagic.compat.waila.provider;
+
+import WayofTime.bloodmagic.api.Constants;
+import WayofTime.bloodmagic.block.BlockBloodTank;
+import WayofTime.bloodmagic.tile.TileBloodTank;
+import WayofTime.bloodmagic.util.helper.TextHelper;
+import com.google.common.base.Strings;
+import mcp.mobius.waila.api.IWailaConfigHandler;
+import mcp.mobius.waila.api.IWailaDataAccessor;
+import mcp.mobius.waila.api.IWailaDataProvider;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+
+import java.util.List;
+
+public class DataProviderBloodTank implements IWailaDataProvider
+{
+    @Override
+    public ItemStack getWailaStack(IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        return null;
+    }
+
+    @Override
+    public List<String> getWailaHead(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        return null;
+    }
+
+    @Override
+    public List<String> getWailaBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        if (!config.getConfig(Constants.Compat.WAILA_CONFIG_BLOOD_TANK))
+            return currenttip;
+
+        if (accessor.getPlayer().isSneaking() || config.getConfig(Constants.Compat.WAILA_CONFIG_BYPASS_SNEAK))
```

Comment:
This whole bit is wrong. You cannot access a Tiles NBT here. Write everything you need to the tag in `#getNBTData` and read it from `accessor#getNBTData`. I need to fix our other implementations to do this. See the altar impl for an example.